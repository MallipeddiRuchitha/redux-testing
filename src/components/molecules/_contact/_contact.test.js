import React from 'react';
import { create } from "react-test-renderer";
import { render ,fireEvent} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Contact from './_contact';
//import {render,fireEvent} from '../../../test-utils'
import { createStore } from 'redux'
import { Provider } from 'react-redux'
 describe("Contact component",() =>{
    const store = createStore(() => ({ contacts: [] }))
    test("Matches the snapshot",() => {
        const input = create(<Provider store={store}><Contact name="abc" email="abc@gmail.com"/></Provider>);
        expect(input.toJSON()).toMatchSnapshot();

    })
    test("Contact element is rendered",() => {
        const { getByTestId, getByText } = render(<Provider store={store}><Contact name="abc" email="abc@gmail.com"/></Provider>);
        expect(getByTestId("contact")).toBeInTheDocument();
        
    });
   
   
   

});