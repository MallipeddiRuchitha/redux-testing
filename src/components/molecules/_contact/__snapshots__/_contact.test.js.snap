// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact component Matches the snapshot 1`] = `
<div
  className="makeStyles-styleContactMainContent-1"
  data-testid="contact"
>
  <p
    className="MuiTypography-root makeStyles-input-2 MuiTypography-body1"
  />
  <p
    className="MuiTypography-root makeStyles-input-2 MuiTypography-body1"
  />
  <div
    className="makeStyles-button-3"
  >
    <div
      data-testid="button"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <span
            className="MuiTypography-root MuiTypography-button"
          >
             
          </span>
        </span>
      </button>
    </div>
  </div>
   
</div>
`;
